src/imap/scan_folders.rs

```diff
@@ -25,7 +25,7 @@ impl Imap {
         }
         info!(context, "Starting full folder scan");
 
-        self.setup_handle(context).await?;
+        self.connect_configured(context).await?;
```

Comment:
connect_configured() -> connect() eventually calls through to try_setup_handle(), so that we only need the latter.

But, why is `setup_handle()` there in the first place? It's called in `Imap::fetch()` and in `idle::idle()`. Both of these are called after a call to `connect_configured()`. Why not always just use `connect_configured()` instead of `setup_handle()`

Reply:
All the `setup_handle`, `unsetup_handle`, `disconnect` and especially `free_connect_param` (now named `free_connect_params`) were there in the C core.

Looks like the whole module could be refactored, like remove `setup_handle`, rename `try_setup_handle` into `setup_handle`, remove `free_connect_params` etc.

Reply:
Refactored `imap` at #2459